pkg/pluginmanager/manager.go

```diff
@@ -989,7 +996,42 @@ func SyncPlugins() error {
 	if err != nil {
 		errList = append(errList, err)
 	}
-	if installedPlugins, err := pluginsupplier.GetInstalledServerPlugins(); err == nil {
+	err = installDiscoveredContextPlugins(plugins)
+	if err != nil {
+		errList = append(errList, err)
+	}
+	return kerrors.NewAggregate(errList)
+}
+
+// SyncPluginsForContextType installs the plugins for given ContextType
+func SyncPluginsForContextType(contextType configtypes.ContextType) error {
+	currentContextMap, err := configlib.GetAllActiveContextsMap()
+	if err != nil {
+		return err
+	}
+	ctx, ok := currentContextMap[contextType]
+	if !ok {
+		return fmt.Errorf("there is no active context for the contextType %v ", contextType)
```

Comment:
Can we use the same format as the same error for `context unset`, so this would be
`"there is no active context for the given context type '%v'"`

Reply:
updated.